File: src/1874/main.js
(() => {
	const readline = require('readline');
	const rl = readline.createInterface({
		input: process.stdin,
		output: process.stdout,
	});
	const buffer = [];
	rl.on('line', (line) => {
		buffer.push(line);
	});
	rl.on('close', () => {
		const input = buffer.map(Number);
		main(input);
	});
})();

const main = ([n, ...input]) => {
	const p = [];
	const q = [];
	let m = 1;

	while (input.length > 0) {
		// console.log('p', p, q);
		const x = input.shift();

		for (let i = m; i <= x; ++i) {
			if (i <= x) {
				p.push(m);
				q.push('+');
				m += 1;
			}
		}
		const y = p.pop();
		if (y !== x) {
			return console.log('NO');
		}
		q.push('-');
	}
	console.log(q.join('\n'));
};

File: src/1874/main.rs
use std::io;

fn main() {
    let mut n = String::new();
    io::stdin().read_line(&mut n).unwrap();
		let n = n.trim().parse::<i32>().unwrap();

		// println!("{}", n);

		let mut m = 1;

		let mut q = Vec::new();
		let mut p = Vec::new();
		for _ in 1..=n {
			let mut s = String::new();
			io::stdin().read_line(&mut s).unwrap();
			let s = s.trim().parse::<i32>().unwrap();

			for i in m..=s {
				p.push(m);
				q.push("+");
				m += 1;
			}

			let t = p.pop().unwrap();
			if(s != t) {
				println!("NO");
				return;
			}
			q.push("-");
		}

		// println!("p {:?}", p);
		// println!("q {:?}", q);

		let x = q.join("\n");
		println!("{}", x);
}
